addons/udes_stock_refactoring/models/stock_picking.py

```diff
@@ -0,0 +1,176 @@
+# -*- coding: utf-8 -*-
+
+import logging
+
+from collections import defaultdict
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    u_mark = fields.Boolean(
+        "Marked",
+        default=True,
+        index=True,
+        help="Pickings that are unused after refactoring are unmarked and deleted",
+    )
+
+    def find_empty_pickings_to_unlink(self):
+        """
+        Finds empty pickings to unlink in self, when it is set, otherwise
+        searches for any empty picking.
+
+        Filter out of the generic search when the picking type does not
+        have auto unlink empty enabled.
+        """
+        PickingType = self.env["stock.picking.type"]
+
+        domain = [
+            ("u_mark", "=", False),
+            ("is_locked", "=", True),
+        ]
+        if self:
+            domain.append(("id", "in", self.ids))
+        else:
+            pts = PickingType.search([("u_auto_unlink_empty", "=", False)])
+            if pts:
+                domain.append(("picking_type_id", "not in", pts.ids))
```

Comment:
this logic seems backwards.

Reply:
Yep, can't think why it was done like this. Maybe the logic was a bit more complex before and got cut down over time.

Anyway have reversed this. :)